File: Features/multirpc-chainmail/workflows/src/integrationTest/java/net/corda/samples/chainmail/FlowTests.java
package net.corda.samples.chainmail;

import com.google.common.collect.ImmutableList;
import net.corda.core.concurrent.CordaFuture;
import net.corda.core.contracts.StateAndRef;
import net.corda.core.node.NetworkParameters;
import net.corda.core.transactions.SignedTransaction;
import net.corda.samples.chainmail.flows.SendMessage;
import net.corda.samples.chainmail.states.MessageState;
import net.corda.testing.node.MockNetwork;
import net.corda.testing.node.MockNetworkParameters;
import net.corda.testing.node.StartedMockNode;
import net.corda.testing.node.TestCordapp;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.time.Instant;
import java.util.Arrays;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.concurrent.ExecutionException;

import static org.junit.Assert.assertEquals;

public class FlowTests {
    private MockNetwork network;
    private StartedMockNode a;
    private StartedMockNode b;
    private StartedMockNode c;

    private final NetworkParameters testNetworkParameters =
            new NetworkParameters(4, Arrays.asList(), 10485760, (10485760 * 5), Instant.now(),1, new LinkedHashMap<>());

    @Before
    public void setup() {
        network = new MockNetwork(new MockNetworkParameters().withCordappsForAllNodes(ImmutableList.of(
                TestCordapp.findCordapp("net.corda.samples.chainmail.contracts"),
                TestCordapp.findCordapp("net.corda.samples.chainmail.flows"))).withNetworkParameters(testNetworkParameters));
        a = network.createPartyNode(null);
        b = network.createPartyNode(null);
        c = network.createPartyNode(null);
        network.runNetwork();
    }

    @After
    public void tearDown() {
        network.stopNodes();
    }

    @Test
    public void MessageFlowTest() throws ExecutionException, InterruptedException {
        String messageText = "Hello, World!";
        SendMessage message = new SendMessage(messageText);
        CordaFuture<SignedTransaction> future1 = a.startFlow(message);
        network.runNetwork();

        for (StartedMockNode node : ImmutableList.of(a, b, c)) {
            node.transaction(() -> {
                List<StateAndRef<MessageState>> messages = node.getServices().getVaultService().queryBy(MessageState.class).getStates();
                assertEquals(1, messages.size());
                MessageState recordedState = messages.get(0).getState().getData();
                assertEquals(recordedState.getMessage(), messageText);
                return null;
            });
        }
    }
}

File: kotlin-samples/Advanced/duediligence-cordapp/workflows/src/test/kotlin/net/corda/samples/duediligence/FlowTests.kt
package net.corda.samples.duediligence

import net.corda.core.contracts.ContractState
import net.corda.core.contracts.StateAndRef
import net.corda.core.contracts.UniqueIdentifier
import net.corda.core.contracts.UniqueIdentifier.Companion.fromString
import net.corda.core.identity.CordaX500Name
import net.corda.core.node.services.Vault.RelevancyStatus
import net.corda.core.node.services.Vault.StateStatus
import net.corda.core.node.services.vault.QueryCriteria
import net.corda.samples.duediligence.flows.RequestToValidateCorporateRecordsInitiator
import net.corda.samples.duediligence.states.CorporateRecordsAuditRequest
import net.corda.testing.node.*
import org.junit.After
import org.junit.Before
import org.junit.Test
import kotlin.test.assertEquals
import java.util.*
import java.util.concurrent.Future


class FlowTests {

    lateinit var network: MockNetwork
    lateinit var a: StartedMockNode
    lateinit var b: StartedMockNode

    @Before
    fun setup() {
        network = MockNetwork(MockNetworkParameters(
                cordappsForAllNodes = listOf(
                        TestCordapp.findCordapp("net.corda.samples.duediligence.contracts"),
                        TestCordapp.findCordapp("net.corda.samples.duediligence.flows")),
                notarySpecs = listOf(MockNetworkNotarySpec(CordaX500Name("Notary","London","GB")))
        ))
        a = network.createPartyNode()
        b = network.createPartyNode()
        network.runNetwork()
    }

    @After
    fun tearDown() {

    }

    @Test
    fun `Auditing Request Creation test`() {
        val flow1 = RequestToValidateCorporateRecordsInitiator(b.info.legalIdentities[0], 10)
        val future = a.startFlow(flow1)
        network.runNetwork()
        val result1 = future.get()
        println(result1)
        val subString = result1?.indexOf("Case Id: ")
        val ApproalID = subString?.plus(9)?.let { result1?.substring(it) }
        println("-$ApproalID-")

        val id = ApproalID?.let { fromString(it) }
        //Query the input
        //Query the input
        val inputCriteria: QueryCriteria.LinearStateQueryCriteria = QueryCriteria.LinearStateQueryCriteria()
                .withUuid(Arrays.asList(UUID.fromString(id.toString())))
                .withStatus(StateStatus.UNCONSUMED)
                .withRelevancyStatus(RelevancyStatus.RELEVANT)
        val (state) = a.services.vaultService.queryBy<ContractState>(ContractState::class.java, inputCriteria).states.get(0)
        val result = state.data as CorporateRecordsAuditRequest
        assertEquals(result.linearId, id)
    }
}